.github/workflows/pdm-lock-automation.yaml

```diff
@@ -76,17 +92,41 @@ jobs:
           else
             echo "Virtual environment already exists."
           fi
+
           source .venv/bin/activate
+
+          echo "Generating/Updating pdm.lock..."
           pdm lock -G :all -v
+
+          echo "Stashing changes if there are any..."
+          git stash || echo "No stash entries found, continuing..."
+
+          echo "Pulling remote ${{ github.head_ref }} branch..."
+          git pull origin ${{ github.head_ref }} --allow-unrelated-histories --no-rebase
+
+          echo "Trying to pop stash if any..."
+          if git stash list | grep -q 'stash@{0}'; then
+            echo "Popping stash..."
+            git stash pop || echo "Failed to pop stash. Continue with other changes."
+          else
+            echo "No stash found, skipping pop."
+          fi
+
           if git diff --quiet; then
             echo "No changes between the current branch and the ${{ github.head_ref }} branch."
             echo "Nothing to commit"
           else
-            git pull origin ${{ github.head_ref }} --allow-unrelated-histories --no-rebase
             echo "Changes detected between the current branch and the ${{ github.head_ref }} branch."
+            echo "Adding pdm.lock changes..."
             git add pdm.lock
+
+            echo "Installing pre-commit..."
             pip install pre-commit~=3.6.2
+
+            echo "Running pre-commit checks..."
             pre-commit run pdm-lock-check
+
+            echo "Committing and pushing changes..."
```

Comment:
@kirtimanmishrazipstack one of the potential issue here is that we might again run in to conflicts by the time we complete pre-commit. I feel like we could skip pre-commit checks in this case. Anyway we have a pre-commit check bot for this. Can you check this with @chandrasekharan-zipstack and @hari-kuriakose

Reply:
@ritwik-g I think pre-commit check will just check the generated pdm lock. We can keep the change.

Reply:
Though we are only performing `lock-check` here the bot very well does this so we can skip the pre-commit check here.

@ritwik-g does the pre-commit check on the PR happen in parallel when this script runs?

Reply:
@ritwik-g  Good call.
However, might be good to continue having a check to validate the generated pdm lock file.

@kirtimanmishrazipstack This pre-commit hook boils down to just `pdm lock --check`.
See https://github.com/pdm-project/pdm/blob/main/.pre-commit-hooks.yaml.

Hence let's not invoke the pre-commit machinery here, instead just run `pdm lock --check` directly.
Should be ok.

Reply:
made the changes @hari-kuriakose